packages/common/src/index.ts

```diff
@@ -1145,7 +1149,11 @@ export default class Common extends EventEmitter {
     if (value) {
       return value
     }
-    return (<any>this._chainParams)['consensus'][this.consensusAlgorithm()]
+    const consensusAlgorithm = this.consensusAlgorithm() as
+      | ConsensusAlgorithm.Ethash
+      | ConsensusAlgorithm.Clique
+      | ConsensusAlgorithm.Casper
```

Comment:
Hmm, I find it a bit unfortunate to hardcode the consensus algorithms here for redundancy reasons. Chances are high that this will be forgotten to be updated once we add a new consensus algorithm (not sure if this is around the corner, but who knows?). Can this be achieved in another way?

Reply:
Good point. I removed this and simplified the typecasting on line 1156. The typecasting is necessary because the TS requires the indexing to be done with the `ConsensusAlgorithm` enum type instead of a string.